lstchain/scripts/lstchain_dvr_pixselector.py

```diff
@@ -0,0 +1,516 @@
+    min_pixel_survival_fraction = Field(np.float32(np.nan), 'min_pixel_survival_fraction')
+    max_pixel_survival_fraction = Field(np.float32(np.nan), 'max_pixel_survival_fraction')
+    mean_pixel_survival_fraction = Field(np.float32(np.nan), 'mean_pixel_survival_fraction')
+    fraction_of_full_shower_events = Field(np.float32(np.nan),
+                                           'fraction_of_full_shower_events')
+    number_of_always_saved_pixels = Field(0, 'number_of_always_saved_pixels')
+    ped_mean = Field(np.float32(np.nan), '')  # photo-electrons
+    ped_stdev = Field(np.float32(np.nan), '') # photo-electrons
+    
+
+def main():
+    args = parser.parse_args()
+    summary_info = RunSummary()
+
+    write_pixel_masks = False
+    if args.action == 'compute_dvr_settings':
+        print('I will calculate the Data Volume Reduction parameters from '
+              'the input DL1 files, and write them to the output file')
+    elif args.action == 'create_pixel_masks':
+        write_pixel_masks = True
+        print('Option to write pixel masks in output file selected. I will '
+              'read in the Data Volume Reduction parameters from the '
+              'corresponding DVR_settings_*.h5 file')
+        print('(the --picture_threshold and --number-of-rings command-line '
+              'options, if provided, will be ignored!)')
+    else:
+        print('Unknown option in --action flag!')
+        return 1
+
+    if args.dl1_files == '':
+        print('Please use --dl1-files provide a valid set of DL1 files!')
+        return 1
+
+    dl1_files = glob.glob(args.dl1_files)
+    dl1_files.sort()
+
+    # The pixel selection will be applied only to "physics triggers"
+    # (showers), not to other events like interleaved pedestal and flatfield
+    # events:
+    event_types_to_be_reduced = [EventType.SUBARRAY.value,
+                                 EventType.UNKNOWN.value,
+                                 EventType.HARDWARE_STEREO.value,
+                                 EventType.DAQ.value]
+    # (Note: some of these types are not yet in use). Chances are that pixels
+    # labeled as "unknown" are cosmics (=showers), so we include them among the
+    # types to be reduced.
+
+    # For the record, keep the input "picture threshold". It is not necessarily
+    # the value that will be used for pixel selection, if it allows too many
+    # pixels too survive.
+    summary_info.picture_threshold = args.picture_threshold
+
+    # Value will be modified, if a too high average fraction (
+    # > max_pix_survival_fraction) of pixels (and FIRST neighbors)
+    # survive, in shower events, with that picture threshold. The value is
+    # computed excluding the 10% noisiest pixels in the camera, because we
+    # want it to be stable when e.g. stars get in and out of the camera,
+    # or when changing wobbles (star field changes)
+    max_pix_survival_fraction = 0.1
```

Comment:
if this is gonna be hardcoded, I would define it at the beginning

Reply:
This is already early inside the main, I put it there because it is related to the charge threshold for pixel selection which is read at that point from the arguments. 
I would not like to make it a configurable parameter, just for simplicity.

Reply:
but are you saying that the same variable is defined twice? this would not be good, but maybe I am misunderstanding... can you point me where is the line you are referring to, please?

Reply:
No, it is just poor wording. picture_threshold is not modified, it is simply the starting value for the actual threshold used, "min_charge_for certain_selection". I will explain it better.